src/h_vialib/client.py

```diff
@@ -58,4 +61,7 @@ def url_for(self, doc):
         options = {"url": doc.url}
         options.update(self.options)
 
+        via_url = self.service_url._replace(path=path, query=urlencode(options))
+        options["via.sec"] = self._secure_token.create(via_url.geturl())
```

Comment:
```suggestion
        options["via.sec"] = self._secure_token.create(doc.url)
```

I think this should be signing the document, rather than our service path. This means the signature is tied to the URL and can be passed on through different end-points, and won't change with different Via options. This also simplifies working out what URL to check on the other side.